Flickr/Flickr/ImageCollectionViewController.swift

```diff
@@ -102,20 +98,27 @@ class ImageCollectionViewController: UIViewController, UICollectionViewDelegate,
                 print("Error parse data")
                 return
         }
-        let photos = Photo.getPhotos(data: photosData)
+        let photos = Photo.getPhotos(from : photosData)
         self.imageData = photos
+        sizes = Photo.getSizes(from: imageData)
+        let laySizes: [CGSize] = sizes.lay_justify(for: view.bounds.size.width, preferredHeight: view.bounds.size.height )
+        sizes = laySizes
         print(data.value ?? "nothing")
         DispatchQueue.main.async() {
-            self.collectionView?.reloadData()
-        }
-    }
-    
-    func getUrlFromArray(photosArray: [Photo], index: Int) -> String? {
-        let photoItem = photosArray[index]
-        guard let urlImage = photoItem.url else {
-            return nil
+            self.searchCollectionView?.reloadData()
+            self.popularCollectionView?.reloadData()
+            if self.isSearching == true {
+                self.searchCollectionView.alpha = 1
```

Comment:
1. search visibility should be triggered by textField focus start/stop. 
2. Only Search visibility should be changing(popularCollectionView should be unchanged)